src/blockdata/script.rs

```diff
@@ -448,9 +457,9 @@ pub struct Instructions<'a> {
 }
 
 impl<'a> Iterator for Instructions<'a> {
-    type Item = Instruction<'a>;
+    type Item = Result<Instruction<'a>, Error>;
 
-    fn next(&mut self) -> Option<Instruction<'a>> {
+    fn next(&mut self) -> Option<Result<Instruction<'a>, Error>> {
```

Comment:
`Option<Result<...>>` is a strange construction as for `Iterator::next()`

For sure, there is a set of `try_*` methods for iterators, returning `Result`, however it's not the `iter()` itself which is supposed to be not only optional, but failable at the second level.

I would probably suggest to move from lazy script parsing to a pre-parsing, returning `Result`, and iterating over it. I can't imagine situation where the script is not consumed as whole, so lazy parsing does not make a lot of sense.

Thus, something like `script.instructions()?.iter().map(|command| ...)` will probably be more idiomatic.

Reply:
Pre-parsing would mean traversing the script twice. I don't know if the speed difference would be noticeable.

At least IO stuff in stdlib does `Item = Result<_>` - https://doc.rust-lang.org/std/io/trait.Read.html#method.bytes , though I must admit handling of such types is under-featured, and I often resolve to using https://docs.rs/resiter/0.4.0/resiter/ .

From PoV of the user I don't find having to handle `Instruction::Error` any better than having to handle `Result` in the `map`. At least with `Item = Result<_>` I can use `resiter`, or write my own reusable adapter, while for `Instruction::Error` it is not possible.

Reply:
Yeah you can either `?` in a loop, or collect if you want the vector
```rust
let instructions: Result<Vec<Instruction<'a>>, Error> = script.instructions().collect();
```
I'm not sure if the following would work:
```rust
let instructions = script.instructions().collect::<Result<Vec<Instruction<'a>>, Error>>().unwrap();
```